File: day-1/puzzle_1.py
import sys
import os
import math

def getFileContent(path):
	fi = open(path, "r")
	content = fi.read()
	fi.close()
	return content

def calcFuel(mass):
	"calculate the amount of fuel necessary for any mass"
	fuel = math.floor(mass / 3) - 2
	return fuel

def main():
	script_directory = sys.path[0]
	input_path = "input.txt" # path relative to the script directory
	full_path = os.path.join(script_directory, input_path)

	aoc_input = getFileContent(full_path)
	mass_list = map(int, aoc_input.split("\n")) # parsing the string for the masses of the modules
	
	fuel_sum = 0
	for mass in mass_list:
		fuel_sum += calcFuel(mass)
	
	print(fuel_sum)
	

if __name__ == "__main__":
	main()

File: day-1/puzzle_1.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <math.h>

using namespace std;

float getFuel(float mass)
{
	float fuel = floor(mass / 3) - 2;
	return fuel;
}

int main()
{
	string file_path = "input.txt";

	string line;
	ifstream input_file(file_path);

	if (input_file.is_open())
	{
		int module_mass;
		float fuel_sum = 0;
		while(getline(input_file, line))
		{
			module_mass = stoi(line);
			fuel_sum += getFuel(module_mass);
		}
		input_file.close();

		cout << fixed << fuel_sum << endl;
	}

	return 0;
}